### DISCUSSION TITLE
iperf3 TCP between openwifi (Xilinx ZYNQ ADI AD9361) board and iPad: 50Mbps

### DISCUSSION THREAD

**Discussion by JiaoXianjun** [open-sdr/openwifi#272] [Show and tell]
https://github.com/open-sdr/openwifi/issues/258

I just tried our latest img with iperf3 between openwifi board and iPad (TCP). It is not that stable, but it does achieve 40~50Mbps with a big chance. Video: https://youtube.com/shorts/0TAqlmuysyo

So my suggestions:
1. Use our latest openwifi img.
2. Use Tx antenna only, and unplug the Rx antenna to give it a "CLEAN" environment from the receiver point of view.
3. Put the client VERY (as close as possible) close to the openwifi board, because openwifi Tx power is much lower than normal WiFi router.
4. IMPORTANT: Find the unused CLEAN WiFi channel at your location/environment (or the least busy channel). 5GHz channel is preferred.
5. Command on openwifi board: "wgd.sh 1" and "fosdem.sh"
6. Command on openwifi board: "iperf3 -s -i1"
7. App on iPad: iPerf 3 Wifi Speed Test (Server address 192.168.13.1, port 5201, Download, 5 Streams, Test duration 10s)

But due to the fact that different boards have different frequency offsets (on board crystal difference), it is normal that if you see the different performance. Our algorithm has limited performance while handling big frequency offset. We are still working on it.

Good luck!

commits:
openwifi https://github.com/open-sdr/openwifi/commit/a0972944154fce74e9db0908970c2bb2f8a31978
openwifi-hw https://github.com/open-sdr/openwifi-hw/commit/691e9d86b788747ed3de74cb2c8793f94b5fff62
openofdm https://github.com/open-sdr/openofdm/commit/747101dca8d00b4d0a99bd07dba647cd49948d04

**Comment by JiaoXianjun** [comment]
This is achieved only while following section does not exist in hostapd configuration file:
```
wpa=1
wpa_passphrase=openwifi
wpa_key_mgmt=WPA-PSK
wpa_pairwise=TKIP CCMP
wpa_ptk_rekey=600
```

**Reply by winfelin** [reply]
The development board we used is MicroPhase e310v2. We tried it according to the method you mentioned above. Whether we measured the speed between two development boards or one development board and iPad, the speed was only about 15Mbps. What is the reason for this?
